src/tests/system/tests/test_sss_override.py

```diff
@@ -0,0 +1,229 @@
+    assert result_getent1.gid == 999011
+    assert result_getent2.uid == 999999
+    assert result_getent2.gid == 888888
+    assert result_getent2.home == "/home/o-user1"
+    assert result_getent3.uid == 999999
+    assert result_getent3.gid == 888888
+    assert result_getent3.home == "/home/o-user1"
+
+
+@pytest.mark.topology(KnownTopology.LDAP)
+@pytest.mark.topology(KnownTopology.AD)
+def test_sss_overrides__overriding_group_name_and_gid(client: Client, provider: GenericProvider):
+    """
+    :title: Locally overriding the name and GID of a group
+    :setup:
+        1. Create group "group1" with posix attributes defined
+        2. Start SSSD
+    :steps:
+        1. Create local override "group1" and name it "o-group1"
+        2. Query groups by the local override name
+        3. Override the GID for the group "group1"
+        4. Query groups by the override name
+    :expectedresults:
+        1. Group local override is created
+        2. Group is found by the overriden name "o-group1"
+        3. Local override POSIX attribute updated
+        4. Group is found by the overriden name "o-group1" and GID changed
+    :customerscenario: False
+    """
+    provider.group("group1").add(gid=999999)
+    client.sssd.domain["ldap_id_mapping"] = "False"
+    client.sssd.config_apply()
+    client.sssd.start()
+    client.sss_override.group("group1").add(name="o-group1")
+    client.sssd.restart()
+
+    result_getent1 = client.tools.getent.group("group1")
+    result_getent2 = client.tools.getent.group("o-group1")
+    client.sss_override.group("group1").add(name="o-group1", gid=888888)
+    result_getent3 = client.tools.getent.group("group1")
+    result_getent4 = client.tools.getent.group("o-group1")
+    client.sss_override.group("group1").delete()
+
+    assert result_getent1
+    assert result_getent1.gid == 999999
+    assert result_getent2
+    assert result_getent3.gid == 888888
+    assert result_getent4.gid == 888888
+
+
+@pytest.mark.topology(KnownTopology.LDAP)
+@pytest.mark.topology(KnownTopology.AD)
+def test_sss_overrides__root_user_cannot_be_overridden(client: Client, provider: GenericProvider):
+    """
+    :title: Root user cannot be overridden
+    :setup:
+        1. Start SSSD
+    :steps:
+        1. Create local override "root" for the user1 user
+    :expectedresults:
+        1. Local override is created but the root user contents are not modified
+    :customerscenario: False
+    """
+    provider.user("user1").add(
+        uid=999011, gid=999011, home="/home/user1", gecos="user", shell="/bin/bash", password="Secret123"
+    )
+    client.sssd.domain["ldap_id_mapping"] = "False"
+    client.sssd.config_apply()
+    client.sssd.start()
+    client.sss_override.user("user1").add(name="root")
+    client.sssd.restart()
+
+    result = client.tools.getent.passwd("root")
+    result_sss = client.tools.getent.passwd("root", service="sss")
+    client.sss_override.user("user1").delete()
+
+    assert not result.uid
```

Comment:
Is it suppose to test that uid is None or uid == 0? Please, use explicit test for readability. Same on other places.

Reply:
It's supposed to make sure that the root UID/GID cannot be used, I updated the test case.